scene/3d/occluder_instance_3d.cpp

```diff
@@ -508,9 +510,9 @@ bool OccluderInstance3D::get_bake_mask_value(int p_layer_number) const {
 }
 
 bool OccluderInstance3D::_bake_material_check(Ref<Material> p_material) {
-	StandardMaterial3D *standard_mat = Object::cast_to<StandardMaterial3D>(p_material.ptr());
-	if (standard_mat && standard_mat->get_transparency() != StandardMaterial3D::TRANSPARENCY_DISABLED) {
-		return false;
+	if (p_material.is_valid() && p_material->get_shader_mode() == Shader::MODE_SPATIAL) {
+		RendererSceneRender *scene_renderer = RendererCompositor::get_singleton()->get_scene();
+		return !scene_renderer->get_material_has_transparency(p_material->get_rid());
```

Comment:
By using this generic way I am independent of the material class (StandardMaterial3D, ShaderMaterial and other that exists or will exist).

I focused on Shader::MODE_SPATIAL because it seems to be the shader for 3D mesh and each RendererSceneRender variant are themselves focusing on this kind of shader.